app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/HorizontalSnapScrollView.kt

```diff
@@ -0,0 +1,269 @@
+
+                    logging.d(LOG_TAG, "column distance: $columnDistance")
+
+                    tempColumnIndex = if (distance > 0) {
+                        activeColumnIndex - columnDistance
+                    } else {
+                        activeColumnIndex + columnDistance
+                    }
+                } else {
+                    if (abs(distance) > columnWidth * MIN_PORTRAIT_SCROLLING_PERCENTAGE) {
+                        tempColumnIndex = if (distance > 0) {
+                            activeColumnIndex - 1
+                        } else {
+                            activeColumnIndex + 1
+                        }
+                    }
+                }
+                scrollToColumn(tempColumnIndex, false)
+                true
+            } else false
+        }
+    }
+
+    override fun onSizeChanged(width: Int, height: Int, oldWidth: Int, oldHeight: Int) {
+        logging.d(LOG_TAG,"onSizeChanged $oldWidth, $oldHeight, $width, $height getMW:$measuredWidth")
+        super.onSizeChanged(width, height, oldWidth, oldHeight)
+
+        displayColumnCount = if (roomsCount == NOT_INITIALIZED) {
+            1
+        } else {
+            calculateDisplayColumnCount(
+                measuredWidth,
+                roomsCount,
+                resources.getInteger(R.integer.max_cols),
+                resources.displayMetrics.density,
+                resources.getInteger(R.integer.min_width_dip)
+            )
+        }
+        val newItemWidth = (measuredWidth.toFloat() / displayColumnCount).roundToInt()
+        logging.d(LOG_TAG, """item width: $newItemWidth ${newItemWidth.toFloat() / resources.displayMetrics.density}dp""")
+        setColumnWidth(newItemWidth)
+    }
+
+    override fun onScrollChanged(scrollX: Int, scrollY: Int, oldScrollX: Int, oldScrollY: Int) {
+        super.onScrollChanged(scrollX, scrollY, oldScrollX, oldScrollY)
+//        logging.d(LOG_TAG, """scrolled from ${oldScrollX} to ${scrollX}""");
+        if (roomNames != null) {
+            roomNames!!.scrollTo(scrollX, 0)
+        }
+    }
+
+    fun setChildScroller(scrollView: HorizontalScrollView) {
+        roomNames = scrollView
+    }
+
+    override fun onLayout(changed: Boolean, left: Int, top: Int, right: Int, bottom: Int) {
+        super.onLayout(changed, left, top, right, bottom)
+        scrollToColumn(activeColumnIndex, true)
+    }
+
+    private fun setColumnWidth(pixels: Int) {
+        logging.d(LOG_TAG, "setColumnWidth $pixels")
+        columnWidth = pixels
+        if (pixels == 0) {
+            return
+        }
+
+        val container = firstChild
+        container.children.forEach { it.layoutParams.width = columnWidth }
+        container.measure(
+            MeasureSpec.makeMeasureSpec(0, MeasureSpec.UNSPECIFIED),
+            MeasureSpec.makeMeasureSpec(0, MeasureSpec.UNSPECIFIED)
+        )
+
+        if (roomNames != null) {
+            roomNames!!.firstChild.children.forEach {
```

Comment:
Did I incorrectly convert this to Kotlin before causing the room names bug?

Reply:
That section caused the bug, but it wasn't a result of your Transpilation. I just made a mistake while copying it 🤦